fontbe/src/features/marks.rs

```diff
@@ -303,6 +307,65 @@ impl<'a> MarkLookupBuilder<'a> {
         }
         result
     }
+
+    fn make_mark_to_liga_lookup(&self) -> Result<Vec<PendingLookup<MarkToLigBuilder>>, Error> {
+        let mut liga_anchor_groups = HashSet::new();
+        // in the future we will probably need to split lookups based on script?
+        // but for now let's just make one lookup.
+        let mut builder = MarkToLigBuilder::default();
+
+        for (glyph_name, anchors) in &self.anchor_lists {
+            let is_not_liga = !self.gdef_classes.is_empty()
+                && (self.gdef_classes.get(glyph_name) != Some(&GlyphClassDef::Ligature));
+            if is_not_liga {
+                continue;
+            }
+            let gid = self.glyph_order.glyph_id(glyph_name).unwrap();
+            let mut component_anchors = Vec::new();
+            for anchor in anchors {
+                if let fontir::ir::AnchorKind::Ligature { group_name, index } = &anchor.kind {
+                    liga_anchor_groups.insert(group_name);
+                    // ensure our component array has an item for this component:
+                    if *index > component_anchors.len() {
+                        component_anchors.resize(*index, BTreeMap::new());
+                    }
+                    let anchor = resolve_anchor(anchor, self.static_metadata, glyph_name)?;
+                    component_anchors[*index - 1].insert(group_name.clone(), anchor);
```

Comment:
I wonder if we should just have a fn or trait to insert at with resize ... if we do this elsewhere as well I think we should. 

I also wonder if it's worth just finding the index of the last ligature and resizing once. I guess we can leave that one for the profiler.

Reply:
on review I agree it's simpler to just figure out the length up front, which also gives us a convenient place to skip a glyph if it doesn't actually have any ligature anchors, despite being classified as a ligature 🤷